### PR TITLE
Exceptions and behaviour while rebasing

### PR DESCRIPTION
**Pull request by k-kryukov**
Updated scripts so that they handle exceptions which might be raised while loading json
Also redirectered stderr to NULL when "git symbolic-ref" is executed so now while rebasing there won't written "fatal: ..."

### CODE REVIEW

File: git_hooks_source_files/prepare-commit-msg [hunk 1]
```diff
@@ -28,38 +31,86 @@ def not_master(branch_name):
...
     try:
-        branch = check_output(["git",
-                               "symbolic-ref",
-                               "--short",
-                               "HEAD"]).decode("utf-8").strip()
+        branch_name = sb.check_output("git symbolic-ref --short HEAD", shell = True)
     except Exception:
-        branch = ""
-        print("HEAD doesn't match any branch tip")
+        print(f"{prefix}[{particular_hook_prefix}]: Error: HEAD is detached")  # mustn't be printed
```
Comment (alexeyandriyash) [inline git_hooks_source_files/prepare-commit-msg:53]: a bit confusing comment

Comment (alexeyandriyash) [reply git_hooks_source_files/prepare-commit-msg:53]: you could just return empty str in this case

**Comment by alexeyandriyash** [comment]
LGTM

**Review by alexeyandriyash — APPROVED** [review]

File: git_hooks_source_files/hooks_tests/hooks_tests.sh [hunk 2]
```diff
@@ -1,59 +1,44 @@
...
 }
 
-function make_comm { # make change and commit
-    echo "\nvoid f() {}" >> ./main.cpp 
-    git add main.cpp
-    git commit -m "$1" >> "$2"
+function test_particular_branch {
+    git checkout "$1"
+    echo "$change_str">> "$test_file_name" # change file somehow
+    comm_res "Some staff on STOR-12344" "$file_path"
```
Comment (alexeyandriyash) [inline git_hooks_source_files/hooks_tests/hooks_tests.sh:22]: just for test consistency better to take pattern from the pattern file

File: git_hooks_source_files/hooks_tests/hooks_tests.sh [hunk 3]
```diff
@@ -1,59 +1,44 @@
...
+    comm_res "Some staff" "$file_path"
+    git reset --hard
 }
 
-change_str="int main() { int* a = 0;*a = 0; }"
+branch_name=$(git symbolic-ref --short HEAD)
 
-git checkout main
 git branch TEST-branch
 git branch TEST-STOR-12345-some-good-news
```
Comment (alexeyandriyash) [inline git_hooks_source_files/hooks_tests/hooks_tests.sh:32]: good news everyone!

Comment (alexeyandriyash) [reply git_hooks_source_files/hooks_tests/hooks_tests.sh:32]: изображение

File: enable_hooks.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+import subprocess as sb
+
+warning_text = "We see, you already use couple of hooks\n" + \
+       "If you want to use our hooks, execute this script with -f flag\n" + \
+       "Note: if -f used, all files from .git/hooks\n" + \
+       "will be turn off by adding .sample to their names"
+hooks_dir = "git_hooks_source_files"
+
+
+def check_if_hooks_exists():
```
[the remark this answers is not in the document]
Comment (k-kryukov) [reply enable_hooks.py:13]: done

**Pull request merged**